### PR TITLE
Fix impassable broken vehicle parts

### LINKED ISSUE #70555: Cant pass through broken car seats and broken trunks
### Describe the bug

Cant pass through broken car seats and broken trunks

### Attach save file

N/A

### Steps to reproduce

Spawn/find broken can with trunks and seats, as example, shooting brake, try to cross trunk/seat - see no way. My character is 185 cm height

### Expected behavior

To have all working as intended 

### Screenshots

_No response_

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19045.3803 (22H2)
- Game Version: 502fa46 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth]
]

### Additional context

_No response_

### LINKED ISSUE #70564: player is too big to fit into the inflatable boat.
### Describe the bug

The player character is unable to enter an inflatable boat due to it not having enough space to fit them, due to the recent changes involving vehicles.

### Attach save file

[Droplets apon Droplets-trimmed.tar.gz](https://github.com/CleverRaven/Cataclysm-DDA/files/13802212/Droplets.apon.Droplets-trimmed.tar.gz)

### Steps to reproduce

1. Place "Inflatable boat"
2. attempt re-entry (your character spawns ontop of the seat, so to get the bug you will have to leave the boat)

### Expected behavior

I should be able to fit into the boat.

### Screenshots

_No response_

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19045.3803 (22H2)
- Game Version: 6bd70ae [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth],
    Mind Over Matter [mindovermatter]
]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by worm-girl**
#### Summary
Bugfixes "Fix impassable broken vehicle parts"

#### Purpose of change

fixes #70555 
fixes #70564 

Vehicle part passability wasn't properly checking for whether certain parts were or were not broken when deciding whether to see if you could move through them. This was my fault!

#### Describe the solution

The check for free cargo space is now bypassed if the part is broken (it can't hold anything if it's broken and shouldn't obstruct anyone based on its contents). The check for cargo_passable also still runs if the part is broken.

Also adjusts some vehicle part sizes that I missed.

#### Describe alternatives you've considered

We could do more in-depth checks on broken parts. Right now they just get a pass as the game sets their capacity to 0 ml which skips the passability check. This allows IE a hulk to smash its way into a car without having to completely delete the tile, though said hulk will still get the cramped space debuff since it's huge. The bucket seat is no longer obstructing the space, but you're still cramming yourself in a too-small vehicle. IMO that makes enough sense that we shouldn't worry about it, but it could be made more granular in the future if desired.

#### Testing

- Stepped on an empty cargo tile.
- Stepped on a broken cargo tile.
- Put a body in a cargo tile, and became unable to step there.
- Stepped on a broken door.
- Stepped on a working door.
- Spawned a zombie and had it do the same.
- Checked that the special case penalties for huge creatures being in vehicles still applied properly in broken parts.
- Checked that inflatable boats are usable again.

### CODE REVIEW

**Comment by prharvey** [comment]
Slightly unrelated to this change, but the cramped_space effect is applied incorrectly for monsters. You're doing it in `monster::will_move_to` which is only a check if a monster should move there, but doesn't actually move the monster. It should be done in `monster::move_to`.

Also note that it's probably not worth spending a lot of time on this, since I'm rewriting the pathfinder anyways: #70274 - I still need to integrate your changes into it.

**Comment by fairyarmadillo** [comment]
> Slightly unrelated to this change, but the cramped_space effect is applied incorrectly for monsters. You're doing it in `monster::will_move_to` which is only a check if a monster should move there, but doesn't actually move the monster. It should be done in `monster::move_to`.
> 
> Also note that it's probably not worth spending a lot of time on this, since I'm rewriting the pathfinder anyways: #70274 - I still need to integrate your changes into it.

Thanks! I moved it to move_to and it seems like it's working properly.

File: src/character.cpp [hunk 1]
```diff
@@ -10906,32 +10905,29 @@ void Character::process_effects()
     }
 
     // Being stuck in tight spaces sucks. TODO: could be expanded to apply to non-vehicle conditions.
-    if( has_effect( effect_cramped_space ) && !in_vehicle ) {
-        remove_effect( effect_cramped_space );
-    }
-    // Check all of this here to ensure the player can't sit in a comfortable seat and then drop 50 liters of junk in their own lap.
-    if( in_vehicle ) {
+    if( has_effect( effect_cramped_space ) ) {
```
Comment (BrettDong) [inline src/character.cpp:10908]: The condition of the if block at line 10943 becomes always `false` after your change at this line, right?

Comment (fairyarmadillo) [reply src/character.cpp:10908]: Oh, it looks like you're right. That shouldn't cause any problems but it should be removed.

File: src/character.cpp [hunk 2]
```diff
@@ -10948,15 +10944,16 @@ void Character::process_effects()
                         add_effect( effect_cramped_space, 2_turns, true );
                     }
                     is_cramped_space = true;
+                    return;
```
Comment (inogenous) [inline src/character.cpp:10947]: These early returns were the cause of #70389 before, which was fixed in #70389 . In this change, the `return`-statements here and on line 10956 are added back. Do you know if this change accidentally re-introduces #70389 again now if you sleep in a cramped space in a vehicle?

Comment (fairyarmadillo) [reply src/character.cpp:10947]: I don't, and they probably do. I didn't realize I was re-introducing the bug.

**Pull request merged**